### PR TITLE
📖 Add release docs to Rancher Turtles repo

### LINKED ISSUE #345: Create release docs
We need to release process documentation for Rancher Turtles.

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it**:
This patch adds a release docs with details on branching, versioning, support and guarantees, release cadence and step-by-step guidance on how to cut a release/push tag and open a PR in this / rancher-turtles-docs repos.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #345 

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [x] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: docs/release.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+        Use existing release branch.
+
+        ```bash
+            # Note: `upstream` must be the remote pointing to `github.com/rancher-sandbox/rancher-turtles`
+            git checkout upstream/release-0.4
+            # Export the tag of the patch release to be cut, e.g.:
+            export RELEASE_TAG=v0.4.1
+        ```
+3. Create a signed, an annotated tag and push it:
```
Comment (salasberryfin) [inline docs/release.md:39]:
```suggestion
3. Create a signed/annotated tag and push it:
```

**Review by salasberryfin — COMMENTED** [review]
Just added a minor nitpick that I copied from CAPRKE2 release docs.

File: docs/release.md (hunk 1 above)
Comment (furkatgofurov7) [reply docs/release.md]: Fixed in both occurences, thanks!

**Review by salasberryfin — DISMISSED** [review]

File: docs/release.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
+# Rancher Turtles Releases
+
+## Release Cadence
+
+- Rancher Turtles minor versions (v0.**4**.0 versus v0.**3**.0) are released every 2-3 weeks or once a month.
+- Rancher Turtles patch versions (v0.4.**2** versus v0.4.**1**) are released as often as weekly.
```
Comment (Danil-Grigorev) [inline docs/release.md:6]:
```suggestion
- New Rancher Turtles versions are released every 2-4 weeks.
```

File: docs/release.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+# Create tags locally
+git tag -s -a ${RELEASE_TAG} -m ${RELEASE_TAG}
+
+# Push tags
+# Note: `upstream` must be the remote pointing to `github.com/rancher-sandbox/rancher-turtles-docs`
+git push upstream ${RELEASE_TAG}
+```
+
+### Create a patch for new version of Rancher Turtles Docs
```
Comment (Danil-Grigorev) [inline docs/release.md:78]: This step is automatic, but this requires to adjust permissions on the repo temporarily here - https://github.com/rancher-sandbox/rancher-turtles-docs/settings/rules/124836. It pushes the version change on behalf of the tag committer.

File: docs/release.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+1. Create a patch with new version of tag (i.e `v0.4`) in Rancher Turtles Docs repository.
```
Comment (Danil-Grigorev) [inline docs/release.md:80]:
```suggestion
1. Ask a team member or directly adjust `docs repo > Settings > Rules > Rulesets > main > Enforcement status > Disabled`
1. Wait for the [workflow](https://github.com/rancher-sandbox/rancher-turtles-docs/blob/main/.github/workflows/version-publish.yaml) to finish
1. Ask a team member or directly adjust `docs repo > Settings > Rules > Rulesets > main > Enforcement status > Active`
```

File: docs/release.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+Prior art: https://github.com/rancher-sandbox/rancher-turtles-docs/commit/d36f0de97c5b5594cd8a396d6780761b6e106e3f
+
+### Deploy site to Pages
+
+1. Run [publish workflow](https://github.com/rancher-sandbox/rancher-turtles-docs/actions/workflows/publish.yaml) manually to
```
Comment (Danil-Grigorev) [inline docs/release.md:86]:
```suggestion
1. Run [publish workflow](https://github.com/rancher-sandbox/rancher-turtles-docs/actions/workflows/publish.yaml) manually to force rollout of updated docs version.
```

**Review by salasberryfin — APPROVED** [review]

**Pull request merged**